tools/approval-service/cmd/approval-service.go

```diff
@@ -114,43 +134,73 @@ func NewGitHubEventSource(processor ApprovalProcessor) *GitHubEventSource {
 // Setup GH client, webhook secret, etc.
 // https://github.com/go-playground/webhooks may help here
 func (ghes *GitHubEventSource) Setup() error {
-	// TODO
+	deployReviewChan := make(chan *github.DeploymentReviewEvent)
+	ghes.deployReviewChan = deployReviewChan
+
+	mux := http.NewServeMux()
+	eventProcessor := webhook.EventHandlerFunc(func(event interface{}) error {
+		switch event := event.(type) {
+		case *github.DeploymentReviewEvent:
+			deployReviewChan <- event
+			return nil
+		default:
+			return trace.Errorf("unknown event type: %T", event)
```

Comment:
[This library only exists for Teleport error handling](https://gravitational.slack.com/archives/C08HF8E9F/p1738272989972819?thread_ts=1738188479.237239&cid=C08HF8E9F), so we probably shouldn't be using it anymore outside of the product (here and elsewhere)